crates/hyperswitch_domain_models/src/payments/payment_intent.rs

```diff
@@ -379,48 +390,192 @@ impl From<PaymentIntentUpdate> for diesel_models::PaymentIntentUpdateInternal {
+                statement_descriptor: None,
+                order_details: None,
+                allowed_payment_method_types: None,
+                metadata: None,
+                connector_metadata: None,
+                feature_metadata: None,
+                enable_payment_link: None,
+                request_incremental_authorization: None,
+                session_expiry: None,
+                frm_metadata: None,
+                request_external_three_ds_authentication: None,
                 updated_by,
             },
-        }
-    }
-}
-
-// This conversion is required for the `apply_changeset` function used for mockdb
-#[cfg(feature = "v2")]
-impl From<PaymentIntentUpdate> for PaymentIntentUpdateInternal {
-    fn from(payment_intent_update: PaymentIntentUpdate) -> Self {
-        match payment_intent_update {
-            PaymentIntentUpdate::ConfirmIntent {
-                status,
-                active_attempt_id,
+            PaymentIntentUpdate::UpdateIntent {
+                amount,
+                currency,
+                shipping_cost,
+                skip_external_tax_calculation,
+                skip_surcharge_calculation,
+                surcharge_amount,
+                tax_on_surcharge,
+                routing_algorithm_id,
+                capture_method,
+                authentication_type,
+                billing_address,
+                shipping_address,
+                customer_present,
+                description,
+                return_url,
+                setup_future_usage,
+                apply_mit_exemption,
+                statement_descriptor,
+                order_details,
+                allowed_payment_method_types,
+                metadata,
+                connector_metadata,
+                feature_metadata,
+                enable_payment_link,
+                request_incremental_authorization,
+                session_expiry,
+                frm_metadata,
+                request_external_three_ds_authentication,
                 updated_by,
             } => Self {
-                status: Some(status),
-                active_attempt_id: Some(active_attempt_id),
-                updated_by,
-                ..Default::default()
-            },
-            PaymentIntentUpdate::ConfirmIntentPostUpdate { status, updated_by } => Self {
-                status: Some(status),
-                updated_by,
-                ..Default::default()
-            },
-            PaymentIntentUpdate::SyncUpdate { status, updated_by } => Self {
-                status: Some(status),
-                updated_by,
-                ..Default::default()
+                status: None,
+                active_attempt_id: None,
+                modified_at: common_utils::date_time::now(),
+
+                amount,
+                currency,
+                shipping_cost,
+                skip_external_tax_calculation: skip_external_tax_calculation.map(|val| match val {
```

Comment:
There is a function to get these as boolean values, maybe you can use them, or implement them if not exists